lms/services/jstor.py

```diff
@@ -1,22 +1,40 @@
+from datetime import timedelta
+
+from lms.services.exceptions import ExternalRequestError
+from lms.services.http import HTTPService
+from lms.services.jwt import JWTService
 from lms.views.helpers import via_url
 
 
 class JSTORService:
-    def __init__(self, enabled, site_code):
+    """An interface for dealing with JSTOR documents."""
+
+    DEFAULT_DOI_PREFIX = "10.2307"
+    """Used when no DOI prefix can be found."""
+
+    # pylint: disable=too-many-arguments
+    def __init__(self, api_url, secret, enabled, site_code, http_service: HTTPService):
         """
         Initialise the JSTOR service.
 
+        :param api_url: JSTOR API url
+        :param secret: Secret for authenticating with JSTOR
         :param enabled: Whether JSTOR is enabled on this instance
         :param site_code: The site code to use to identify the organization
+        :param http_service: HTTPService instance
         """
+        self._api_url = api_url
+        self._secret = secret
         self._enabled = enabled
         self._site_code = site_code
 
+        self._http = http_service
+
     @property
     def enabled(self) -> bool:
         """Get whether this instance is configured for JSTOR."""
 
-        return bool(self._enabled and self._site_code)
+        return bool(self._enabled and self._api_url and self._site_code)
```

Comment:
As this type of service/sources proliferate we might want to also make these checks on the admin pages, ie "you enabled this but it's missing the site code and the env var is not set."